### PR TITLE
Link check

### LINKED ISSUE #1476: Add hyperlink checking in CI
Add hyperlink checking to [the docs GH workflow](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/.github/workflows/validate-documentation.yml).

We can use https://github.com/lycheeverse/lychee-action

### PR DESCRIPTION
**Pull request by Kielek**
## Why

Fixes #1476

## What

Link check in MD files.
Fixes for broken links.

## Tests

CI

## Checklist

- ~~[ ] `CHANGELOG.md` is updated.~~
- [x] Documentation is updated.
- ~~[ ] New features are covered by tests.~~

### CODE REVIEW

**Review by RassK — APPROVED** [review]
👍

File: .github/workflows/validate-documentation.yml [hunk 1]
```diff
@@ -7,19 +7,30 @@ on:
...
     - '**.md'
     - 'package-lock.json'
     - 'package.json'
+    - '.lycheeignore'
+  workflow_dispatch:
+    paths:
+    - '**.md'
+    - 'package-lock.json'
+    - 'package.json'
+    - '.lycheeignore'
```
Comment (pellared) [inline .github/workflows/validate-documentation.yml:23]: why have you add `paths` to `workflow_dispatch`? Does it do anything? I think this should be removed.

**Review by pellared — COMMENTED** [review]
Awesome 👍 Just one comment to double-check.

File: .github/workflows/validate-documentation.yml (hunk 1 above)
Comment (Kielek) [reply .github/workflows/validate-documentation.yml]: c77071a9ccf2e26dd341e65937a37e8b1dd11d22

**Review by pellared — APPROVED** [review]

**Comment by Kielek** [comment]
Warning in document validation should be fixed with new lyche-action: https://github.com/lycheeverse/lychee-action/issues/168

**Pull request merged**